magicctapipe/scripts/lst1_magic/lst1_magic_dl1_stereo_to_dl2.py

```diff
@@ -2,82 +2,266 @@
+    # event. In order to speed up the calculations, here we separate the
+    # input events by the telescope combination types.
+
+    reco_params = pd.DataFrame()
+
+    for tel_ids in TEL_COMBINATIONS.values():
+
+        multiplicity = 2 * len(tel_ids)
+
+        df_events = reco_params_flips.query(
+            f"(multiplicity == {multiplicity}) & (tel_id == {tel_ids})"
+        ).copy()
+
+        df_events["multiplicity"] = df_events.groupby(["obs_id", "event_id"]).size()
+        df_events.query(f"multiplicity == {multiplicity}", inplace=True)
+
+        n_events = len(df_events.groupby(["obs_id", "event_id"]).size())
+
+        # Here we first define all the possible flip combinations. For
+        # example, in case that we have two telescope images, in total
+        # 4 combinations can be defined as follows:
+        #   [(head, head), (head, tail), (tail, head), (tail, tail)]
+        # where the i-th element of each tuple means the i-th telescope
+        # image. In case of 3 images we have in total 8 combinations.
+
+        flip_combinations = np.array(
+            list(itertools.product([0, 1], repeat=len(tel_ids)))
+        )
+
+        # Next, we define all the possible any 2 telescope combinations.
+        # For example, in case of 3 telescopes, in total 3 combinations
+        # are defined as follows:
+        #                 [(1, 2), (1, 3), (2, 3)]
+        # where the elements of the tuples mean the telescope IDs. In
+        # case of 2 telescopes there is only one combination.
+
+        tel_any2_combinations = list(itertools.combinations(tel_ids, 2))
+
+        distances = np.zeros((len(flip_combinations), n_events))
+
+        for i_flip, flip_combo in enumerate(flip_combinations):
+
+            container = {}
+
+            # Set the directions of a given flip combination
+            for tel_id, flip in zip(tel_ids, flip_combo):
+                container[tel_id] = df_events.query(
+                    f"(tel_id == {tel_id}) & (flip == {flip})"
+                )
+
+            for tel_any2_combo in tel_any2_combinations:
+
+                tel_id_1 = tel_any2_combo[0]
+                tel_id_2 = tel_any2_combo[1]
+
+                theta = angular_separation(
+                    lon1=u.Quantity(container[tel_id_1]["reco_az"].to_numpy(), u.deg),
+                    lat1=u.Quantity(container[tel_id_1]["reco_alt"].to_numpy(), u.deg),
+                    lon2=u.Quantity(container[tel_id_2]["reco_az"].to_numpy(), u.deg),
+                    lat2=u.Quantity(container[tel_id_2]["reco_alt"].to_numpy(), u.deg),
+                )
+
+                # Sum up the distance per flip combination
+                distances[i_flip] += theta.to_value(u.deg)
+
+        # Finally, we extract the indices of the flip combinations for
+        # each event with which the angular distances become minimum
+        distances = np.array(distances)
+        distances_min = distances.min(axis=0)
```

Comment:
maybe you can use np.argmin

Reply:
Thank you, I did not notice this function, which is useful actually, so I changed the code.